mergify_engine/config.py

```diff
@@ -325,56 +325,65 @@ def ApplicationAPIKeys(v: str) -> typing.Dict[str, ApplicationAPIKey]:
 TESTING_INSTALLATION_ID: int
 SAAS_MODE: bool
 
-configuration_file = os.getenv("MERGIFYENGINE_TEST_SETTINGS")
+# config variables built
+GITHUB_DOMAIN: str
 
-if configuration_file is not None:
-    dotenv.load_dotenv(dotenv_path=configuration_file, override=True)
 
-CONFIG: typing.Dict[str, typing.Any] = {}
-for key, _ in Schema.schema.items():
-    val = os.getenv(f"MERGIFYENGINE_{key}")
-    if val is not None:
-        CONFIG[key] = val
+def load() -> typing.Dict[str, typing.Any]:
+    configuration_file = os.getenv("MERGIFYENGINE_TEST_SETTINGS")
 
+    if configuration_file is not None:
+        dotenv.load_dotenv(dotenv_path=configuration_file, override=True)
 
-# DASHBOARD API KEYS are required only for Saas
-if "SUBSCRIPTION_TOKEN" in CONFIG:
-    for key in ("DASHBOARD_TO_ENGINE_API_KEY", "ENGINE_TO_DASHBOARD_API_KEY"):
-        CONFIG[key] = secrets.token_hex(16)
+    raw_config: typing.Dict[str, typing.Any] = {}
+    for key, _ in Schema.schema.items():
+        val = os.getenv(f"MERGIFYENGINE_{key}")
+        if val is not None:
+            raw_config[key] = val
 
-legacy_api_url = os.getenv("MERGIFYENGINE_GITHUB_API_URL")
-if legacy_api_url is not None:
-    if legacy_api_url[-1] == "/":
-        legacy_api_url = legacy_api_url[:-1]
-    if legacy_api_url.endswith("/api/v3"):
-        CONFIG["GITHUB_REST_API_URL"] = legacy_api_url
-        CONFIG["GITHUB_GRAPHQL_API_URL"] = f"{legacy_api_url[:-3]}/graphql"
+    # DASHBOARD API KEYS are required only for Saas
+    if "SUBSCRIPTION_TOKEN" in raw_config:
+        for key in ("DASHBOARD_TO_ENGINE_API_KEY", "ENGINE_TO_DASHBOARD_API_KEY"):
+            raw_config[key] = secrets.token_hex(16)
```

Comment:
This is confusing because I imagine this does not work then and is useless?
Might be better to use a static string like `unused`?

Reply:
The thing is that the web servers' endpoints are always mounted, so I put random secrets so nobody can use them on-premise.

Known that we have `SAAS_MODE` boolean I thing we can do better. I created a ticket to fix this.